### DISCUSSION TITLE
Pushing scripts to blade stacks not working from step components.

### DISCUSSION THREAD

**Discussion by reppair** [spatie/laravel-livewire-wizard#35] [Q&A] [answered]
Hi there, I have a form in a step component that uses Google reCaptcha, and I want to push the reCaptcha script from the step template so it gets loaded when the step component's template is rendered. Is that possible? 

This works
```blade
// parent component template

<livewire:wizard-components />

@push('scripts')
<script>alert('It works!')</script>
@endpush
```

but this won't work
```blade
// step component template

@push('scripts')
<script>alert('I hoped it would work.')</script>
@endpush
```

**Comment by ju5t** [comment, marked as answer]
You can't use `@push` in dynamically rendered components in Livewire as far as I know. This is not a limitation of this package but it's how Livewire works. As Livewire uses dom diff, it can't push data into a stack that's already rendered.

**Reply by reppair** [reply]
Docs: https://laravel-livewire.com/docs/2.x/inline-scripts#introduction

> You can also push scripts directly onto Blade stacks from your Livewire component

What exactly do you mean by 'dynamically rendered', like when used in an `if` statement?
```blade
@if($authFlow == 'register')
    <livewire:organization.auth.onboarding.onboarding-wizard />
@else
    <livewire:organization.auth.login />
@endif
```

**Reply by reppair** [reply]
Kind of makes sense what you are saying

> it can't push data into a stack that's already rendered

hmmm.. 🤔

**Reply by ju5t** [reply]
> What exactly do you mean by 'dynamically rendered', like when used in an if statement?

You can only use `@push` the first time the page fully renders. `@push` does not work in any of the subsequent requests. As the wizard renders each step without a page refresh, `@push` will not have access to the stack of your base layout view.
